tests/test_pylint_datetime.py

```diff
@@ -2,7 +2,308 @@
+
+        Args:
+            msg_id: message id.
+            node: node to check.
+            node2: second node to check.
+            args: arguments to pass to the message.
+        """
+        with self.assertAddsMessages(
+            pylint.testutils.MessageTest(
+                msg_id=msg_id,
+                node=node if node2 is None else node2,
+                args=args,
+            ),
+            ignore_position=True,
+        ):
+            self.checker.visit_assign(node)
+
+    def test_timedelta_without_kargs(self) -> None:
+        """Test timedelta without keyword arguments."""
+        call_node_1, call_node_2 = astroid.extract_node(
+            """
+        import datetime
+        from datetime import timedelta
+        timedelta(1) #@
+        datetime.timedelta(1) #@
+        """
+        )
+        self.asserting_messages_calls("timedelta-no-keyword-args", call_node_1)
+        self.asserting_messages_calls("timedelta-no-keyword-args", call_node_2)
+
+    def test_timedelta_with_kargs(self) -> None:
+        """Test timedelta with keyword arguments."""
+        call_node_1, call_node_2 = astroid.extract_node(
+            """
+        import datetime
+        from datetime import timedelta
+        timedelta(seconds=1) #@
+        datetime.timedelta(seconds=1) #@
+        """
+        )
+        with self.assertNoMessages():
+            self.checker.visit_call(call_node_1)
+            self.checker.visit_call(call_node_2)
+
+    def test_no_timezone_argument(self) -> None:
+        """Test missing timezone arguments."""
+        (
+            call_node_1,
+            call_node_2,
+            call_node_3,
+            call_node_4,
+        ) = astroid.extract_node(
+            """
+        import datetime
+        from datetime import datetime
+        datetime.now() #@
+        datetime.fromtimestamp(1)  #@
+        datetime(1, 1, 1, 1, 1, 1, 1)  #@
+        datetime.astimezone()  #@
+        """
+        )
+        self.asserting_messages_calls(
+            "timezone-no-argument", call_node_1, args=("now",)
+        )
+        self.asserting_messages_calls(
+            "timezone-no-argument", call_node_2, args=("fromtimestamp",)
+        )
+        self.asserting_messages_calls(
+            "timezone-no-argument", call_node_3, args=("datetime",)
+        )
+        self.asserting_messages_calls(
+            "timezone-no-argument", call_node_4, args=("astimezone",)
+        )
+
+    def test_passed_timezone_argument(self) -> None:
+        """Test correctly passed timezone arguments."""
+        (
+            call_node_1,
+            call_node_2,
+            call_node_3,
+            call_node_4,
+        ) = astroid.extract_node(
+            """
+        import datetime
+        from datetime import datetime, timezone
+        datetime.now(tz=timezone.utc) #@
+        datetime.fromtimestamp(1, timezone.utc)  #@
+        datetime(1, 1, 1, 1, 1, 1, 1, timezone.utc)  #@
```

Comment:
Probably for a different PR, but I wonder if we also want to forbid calling `datetime()` without keyword arguments, but maybe is not as unclear as `timedelta`.